scantool/diag_cfg.c

```diff
@@ -51,8 +52,9 @@ int diag_cfg_setstr(struct cfgi *cfgp, const char *str) {
 		free(cfgp->val.str);
 		cfgp->val.str = NULL;
 	}
-	if (diag_malloc(&cfgp->val.str, slen+1)) {
-		return diag_iseterr(DIAG_ERR_NOMEM);
+	rv = diag_malloc(&cfgp->val.str, slen+1);
+	if (rv != 0) {
```

Comment:
PS I'm also ok with having the `if (rv) {` form, it's pretty unambiguous . Either way is fine.

Reply:
Well it is conceptually cleaner to me (and would be to you too if you got used to it P) ) to only have boolean values in selection and iteration statement checks.

Reply:
Obviously, no need to refactor the whole codebase to unify these, this particular inconsistency is quite tolerable !